### ISSUE TITLE
Area chart gradient becoming black in Safari

### ISSUE THREAD

**Issue by Gunner17** [recharts/recharts#1813]
Area chart gradient becoming black in Safari (version 12.0.3).

Here is the AreaChart definition with LinearGradient:
```
<AreaChart
	data={transformedData}
	margin={{
		right: 100, left: 0, top: 10, bottom: 10,
	}}
>
	<defs>
		<linearGradient id="colorUv" x1="0" y1="0" x2="0" y2="1">
			<stop offset="5%" stopColor="#325b98" stopOpacity={0.95} />
			<stop offset="95%" stopColor="#aacbeb" stopOpacity={0.8} />
		</linearGradient>
	</defs>
...
```

Here is the Area component:

```
<Area
	isAnimationActive={false}
	stroke={colors[index]}
	type="monotone"
	fillOpacity={1}
	dataKey={series.name}
	key={series.name}
	fill={config.data.series.length > 1 ? colors[index] : 'url(#colorUv)'}
/>
```

This is the result in Chrome:
<img width="971" alt="Screen Shot 2019-07-23 at 12 28 42 AM" src="https://user-images.githubusercontent.com/11354728/61691932-7f027f80-ace1-11e9-8f4b-a4186b92d51a.png">

But, it goes black in Safari:
<img width="817" alt="Screen Shot 2019-07-23 at 12 34 50 AM" src="https://user-images.githubusercontent.com/11354728/61692024-ba9d4980-ace1-11e9-9e7d-4c6df4fd8294.png">

Any idea, what I could be doing wrong? Thanks in advance.

**Comment by edferreira** [comment]
the same is happening here

**Comment by edferreira** [comment]
I solved something similar to this removing my defs to an <svg/> tag outside of <AreaChart/>

**Comment by nikdementev** [comment]
@edferreira Please help me with this problem)

**Comment by kunalkp** [comment]
Getting the same issue on safari for bar graph, only when I refresh. I can see in inspect that there's no defs once I refresh the page.

**Comment by wrenzi** [comment]
Hey folks. I solved a similar Issue, and I'll share the problem/solution with you.

The problem, in our case, was not with the recharts themselves, but with our implementation.

We were adding two graphs to our page, with their respectives `defs`, and the `defs` were duplicating (two `defs` with the same id). The issue happened when we removed the last `linearGradient` def, and kept the first one. It made our `AreaChart` lose the fill def reference, even though the first one still exists.

The solution was to create `linearGradient` with different ids.

I hope it helps someone.
